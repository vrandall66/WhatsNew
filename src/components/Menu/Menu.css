.Menu {
  color: #ffffff;
  padding: 0% 2%;
  width: 20%;
}

.Menu-header {
  font-size: 2.15em;
  padding: 10% 2%;
}

.Menu-header-span {
  color: #ffae03;
}

.Menu-ul {
  list-style: none;
  font-size: 1.5em;
}

.Menu-li {
  align-items: center;
  display: flex;
  margin: 10% 2% 18% 2%;
}

.Menu-li-span {
  font-size: 1.5em;
  margin-right: 10%;
}

.Menu-li:hover,
.Menu-li-span:hover {
  background-color: #ffae03;
  border-radius: 0px 75px 75px 0px;
  cursor: pointer;
}
