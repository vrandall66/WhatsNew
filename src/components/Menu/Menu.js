import React from "react";
import { TiLocationOutline } from "react-icons/ti";
import { FaCogs } from "react-icons/fa";
import { GiTicket } from "react-icons/gi";
import { GiMicroscope } from "react-icons/gi";
import { FaHeartbeat } from "react-icons/fa";
import "./Menu.css";

const Menu = ({ changeData }) => {
  return (
    <aside className="Menu">
      <h1 className="Menu-header">
        What's <span className="Menu-header-span">New?</span>
      </h1>
      <ul className="Menu-ul">
        <li className="Menu-li" id="local" onClick={(event) => changeData(event)}>
          <span className="Menu-li-span">
            <TiLocationOutline />
          </span>
          Local News
        </li>
        <li className="Menu-li" id="technology" onClick={(event) => changeData(event)}>
          <span className="Menu-li-span">
            <FaCogs />
          </span>
          Technology
        </li>
        <li className="Menu-li" id="entertainment" onClick={(event) => changeData(event)}>
          <span className="Menu-li-span">
            <GiTicket />
          </span>
          Entertainment
        </li>
        <li className="Menu-li" id="science" onClick={(event) => changeData(event)}>
          <span className="Menu-li-span">
            <GiMicroscope />
          </span>
          Science
        </li>
        <li className="Menu-li" id="health" onClick={(event) => changeData(event)}>
          <span className="Menu-li-span">
            <FaHeartbeat />
          </span>
          Health
        </li>
      </ul>
    </aside>
  );
};

export default Menu;
