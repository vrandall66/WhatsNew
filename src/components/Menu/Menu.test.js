import React from "react";
import { shallow } from "enzyme";
import Menu from "./Menu";

describe("Menu", () => {
  let wrapper;
  const mockChangeData = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Menu changeData={mockChangeData} />);
  });

  it("Should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
