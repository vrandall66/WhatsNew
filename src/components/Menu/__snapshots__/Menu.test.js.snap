// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Should match snapshot 1`] = `
<aside
  className="Menu"
>
  <h1
    className="Menu-header"
  >
    What's 
    <span
      className="Menu-header-span"
    >
      New?
    </span>
  </h1>
  <ul
    className="Menu-ul"
  >
    <li
      className="Menu-li"
      id="local"
      onClick={[Function]}
    >
      <span
        className="Menu-li-span"
      >
        <TiLocationOutline />
      </span>
      Local News
    </li>
    <li
      className="Menu-li"
      id="technology"
      onClick={[Function]}
    >
      <span
        className="Menu-li-span"
      >
        <FaCogs />
      </span>
      Technology
    </li>
    <li
      className="Menu-li"
      id="entertainment"
      onClick={[Function]}
    >
      <span
        className="Menu-li-span"
      >
        <GiTicket />
      </span>
      Entertainment
    </li>
    <li
      className="Menu-li"
      id="science"
      onClick={[Function]}
    >
      <span
        className="Menu-li-span"
      >
        <GiMicroscope />
      </span>
      Science
    </li>
    <li
      className="Menu-li"
      id="health"
      onClick={[Function]}
    >
      <span
        className="Menu-li-span"
      >
        <FaHeartbeat />
      </span>
      Health
    </li>
  </ul>
</aside>
`;
